tests/scripts/generate_tls13_psk_tests.py

```diff
@@ -0,0 +1,522 @@
+# limitations under the License.
+
+"""
+Generate TLSv1.3 PSK test cases
+
+"""
+
+import sys
+import os
+import argparse
+import itertools
+from collections import namedtuple
+
+# define certificates configuration entry
+Certificate = namedtuple("Certificate", ['cafile', 'certfile', 'keyfile'])
+# define the certificate parameters for signature algorithms
+CERTIFICATES = {
+    'ecdsa_secp256r1_sha256': Certificate('data_files/test-ca2.crt',
+                                          'data_files/ecdsa_secp256r1.crt',
+                                          'data_files/ecdsa_secp256r1.key'),
+    'ecdsa_secp384r1_sha384': Certificate('data_files/test-ca2.crt',
+                                          'data_files/ecdsa_secp384r1.crt',
+                                          'data_files/ecdsa_secp384r1.key'),
+    'ecdsa_secp521r1_sha512': Certificate('data_files/test-ca2.crt',
+                                          'data_files/ecdsa_secp521r1.crt',
+                                          'data_files/ecdsa_secp521r1.key'),
+    'rsa_pss_rsae_sha256': Certificate('data_files/test-ca_cat12.crt',
+                                       'data_files/server2-sha256.crt', 'data_files/server2.key'
+                                       )
+}
+
+MBEDTLS_KEX_MODES = {
+    'psk': 'psk',
+    'psk_ephemeral': 'psk_ephemeral',
+    'psk_all': 'psk_all',
+    'ephemeral': 'ephemeral',
+    'ephemeral_all': 'ephemeral_all',
+    'all': 'all',
+}
+
+OPENSSL_KEX_MODES = {
+    'psk_or_ephemeral': 'psk_or_ephemeral',
+    'ephemeral': 'ephemeral',
+    'all': 'all',
+}
+
+GNUTLS_KEX_MODES = {
+    'psk_or_ephemeral': 'psk_or_ephemeral',
+    'ephemeral': 'ephemeral',
+    'ephemeral_all': 'ephemeral_all',
+    'all': 'all',
+}
+
+SERVER_TO_MODE = {
+    'mbedTLS': MBEDTLS_KEX_MODES,
+    'OpenSSL': OPENSSL_KEX_MODES,
+    'GnuTLS': GNUTLS_KEX_MODES
+}
+
+PSK_SETS = {
+    'good_match': 'good',
+    'id_mismatch': 'fail,no common id',
+    'psk_mismatch': 'fail,no common psk',
+}
+
+MBEDTLS_PSK_IDENTITIES = {
+    'good_match': 'Client_Identity',
+    'id_mismatch': 'Wrong_Identity',
+    'psk_mismatch': 'Client_Identity',
+}
+
+MBEDTLS_PSK_MATERIALS = {
+    'good_match': '6162636465666768696a6b6c6d6e6f70',
+    'id_mismatch': '6162636465666768696a6b6c6d6e6f70',
+    'psk_mismatch': '6162636465666768696a6b6c6d6e6f71',
+}
+
+OTHER_PSK_IDENTITIES = {
+    'good_match': 'Client_Identity',
+}
+
+OTHER_PSK_MATERIALS = {
+    'good_match': '6162636465666768696a6b6c6d6e6f70',
+}
+
+MSG_TO_ID = {
+    'mbedTLS': MBEDTLS_PSK_IDENTITIES,
+    'OpenSSL': OTHER_PSK_IDENTITIES,
+    'GnuTLS': OTHER_PSK_IDENTITIES
+}
+
+MSG_TO_MATERIAL = {
+    'mbedTLS': MBEDTLS_PSK_MATERIALS,
+    'OpenSSL': OTHER_PSK_MATERIALS,
+    'GnuTLS': OTHER_PSK_MATERIALS
+}
+
+PSK_SUB_MESSAGES = {
+    'good_match': 'found psk key exchange modes extension',
+    'id_mismatch': 'No matched PSK or ticket',
+    'psk_mismatch': 'Invalid binder.',
+}
+
```

Comment:
In expected modes calculating, I use python SET type. It can be replace with integer within this comment
```suggestion
#CONST value for calculating expected kex modes 
KEY_EXCHANGE_MODES=dict(psk=1, ephemeral=2, psk_epemeral=4, psk_or_ephemeral=3, psk_all=5, ephemeral_all=6, all=7)
```